File: app/src/main/java/com/danielspeixoto/connect/model/WebService/UsersService.java
package com.danielspeixoto.connect.model.WebService;

import com.danielspeixoto.connect.model.pojo.User;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

/**
 * Created by daniel on 16/04/17.
 */

public interface UsersService {
	
	@GET("users")
	Observable<List<User>> getUsers();
	
	@POST("users")
	Observable<User> createUser(@Body User user);
	
}

File: app/src/main/java/com/danielspeixoto/connect/model/WebService/UsersService.kt
package com.danielspeixoto.connect.model.WebService

import com.danielspeixoto.connect.model.UserModel
import com.danielspeixoto.connect.model.pojo.User
import com.danielspeixoto.connect.model.pojo.Visitor
import retrofit2.Call
import retrofit2.http.*

/**
 * Created by danielspeixoto on 4/21/17.
 */

interface UsersService {

    @POST("users")
    fun createADM(@Body user: User): Call<User>

    @POST("users/{group}")
    fun createWorker(@Header("Authorization") authorization: String,
                     @Path("group") group: String,
                     @Body user: User): Call<User>

    @PUT("users/login")
    fun logIn(@Body user: User): Call<User>

    @GET("users/{group}")
    fun getCoWorkers(@Header("Authorization") authorization: String,
                     @Path("group") group: String): Call<List<User>>

    @GET("users/{id}/visitors")
    fun getVisitors(@Header("Authorization") authorization: String = UserModel.currentUser!!.token!!,
                    @Path("id") username: String = UserModel.currentUser!!.username!!) : Call<List<Visitor>>
}